source/analyze-code.txt

```diff
@@ -0,0 +1,224 @@
+.. _visual-studio-extension-builders:
+
+=================
+Analyze Your Code
+=================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Learn how to use the {+product+} to analyze your {+driver-long+} code.
+The {+product+} can analyze **builder** and **Language Integrated Query (LINQ)** expressions.
+
+A builder is a class provided by the {+driver-short+} to help you construct
+common operations like queries and updates.
+
+LINQ is a query syntax included in the C# language. The {+driver-short+}
+can translate a subset of LINQ expressions into MongoDB aggregation pipelines.
+
+To learn more about builders, see 
+`Builders <{+driver-docs+}reference/driver/definitions/>`__ in the
+{+driver-short+} documentation.
+
+To learn more about LINQ, see the following resources:
+
+- `LINQ <{+driver-docs+}reference/driver/crud/linq/>`__ in the {+driver-short+}
+  documentation
+- `LINQ <https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/>`__
+  in the Microsoft C# guide
+
+To learn more about aggregation pipelines, see
+:manual:`Aggregation </aggregation/>` in the MongoDB manual.
+
+Analyze Builders
+----------------
+
+Use the {+product+} to translate your builder expressions into the {+query-api+}.
+Click the following tabs to see an example of a builder expression
+and its corresponding {+query-api+} translation:
+
+.. tabs::
+
+   .. tab:: Builders
+      :tabid: builder
+
+      .. code-block:: csharp
+
+         var filter = Builders<Movie>.Filter.Eq(m => m.Genre, genre) &
```

Comment:
minor: Should the new lines alignment match the linq sample code:
```
var filter = Builders<Movie>.Filter.Eq(m => m.Genre, genre) &
  Builders<Movie>.Filter.Gte(m => m.Score, minScore) &
  Builders<Movie>.Filter.Regex(m => m.Score, titleSearchTerm);
```

Reply:
Good catch, It seems like the builder sample code [aligns with the variable name ](https://github.com/BorisDog/mongo-csharp-analyzer/blob/vs20/samples/BasicSample/BuildersSample.cs#L15-L18) as well as [the LINQ sample code](https://github.com/BorisDog/mongo-csharp-analyzer/blob/vs20/samples/BasicSample/LinqSample.cs#L34-L37).